### PR TITLE
feat: Add global search bar

### LINKED ISSUE #1578: feat: 🔍 Add Search bar to Docs site
# 🙋 Feature Request

> [!NOTE]  
> I am happy to implement MVP solution but want sign off & a point in the right direction first as a first contribution :)

There are a lot of component pages & articles in the docs site & it is sometimes hard to jump to the correct component.
A search bar/button at the top would assist greatly

image
**Figure: Great spot for a search button**

## 🤔 Expected Behavior

MVP: Search Nav Menu links
Non-MVP: Search page content as well

## 😯 Current Behavior

No search ability, beyond control + f, but sometimes menus are collapsed :(

## 💁 Possible Solution

MVP: Some injected menu, & string search the entries
Non-MVP: More thought to be had, do we use a 3rd party tool?

## 🔦 Context

As a developer working with Fluent UI Blazor
I want to navigate the docs site quickly
So that I can develop my solutions

## 💻 Examples

image
Figure: https://mudblazor.com/

image
Figure: https://antblazor.com/

### PR DESCRIPTION
**Pull request by Hona**
# Pull Request

## 📖 Description

from

image

to

image

Example when typing valid content:

image

Note: Hides on mobile - future work could make it nice & responsive. Probably an Icon only button that pops up a fullscreen overlay

image

### 🎫 Issues

Fixes #1578 

## 📑 Test Plan

Manually test the nav menu looks the same (I tested each item 1 by 1 & saw no difference).
Test that the search bar behaves as a nice UX.

### General

~~- [ ] I have added tests for my changes.~~
Not done - No tests in Demo site
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Comment by Hona** [comment]
@microsoft-github-policy-service agree

File: examples/Demo/Shared/Shared/DemoSearch.razor [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                <span slot="start">
+                    <FluentIcon Value="@(item.Icon)" Class="search-result-icon" Color="Color.Neutral" Slot="start"/>
+                </span>
+                @item.Title
+            </FluentMenuItem>
+        }
+    </FluentMenu>
+}
+
+@code {
```
Comment (dvoituron) [inline examples/Demo/Shared/Shared/DemoSearch.razor:30]: If it is possible, could you separate the razor and the code parts (in a .cs file)? Thanks a lot :-)

File: examples/Demo/Shared/Shared/DemoSearch.razor [hunk 2]
```diff
@@ -0,0 +1,82 @@
+﻿@using System.Diagnostics
+@inject DemoNavProvider NavProvider
+@inject NavigationManager NavigationManager
+
+<FluentSearch @bind-Value="@_searchValue"
+              Immediate
+              @bind-Value:after=HandleSearchInput
```
Comment (dvoituron) [inline examples/Demo/Shared/Shared/DemoSearch.razor:7]: could you enclose all the attributes in double quotes?

File: examples/Demo/Shared/Shared/DemoSearch.razor [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            _searchResults = DefaultResults();
+            _searchValue = string.Empty;
+        }
+        else
+        {
+            var searchTerm = _searchValue.ToLower();
+
+            if (searchTerm.Length > 0)
+            {
+                var temp = NavProvider.FlattenedMenuItems
```
Comment (dvoituron) [inline examples/Demo/Shared/Shared/DemoSearch.razor:58]: Could you find an explicit name, in place of "temp" ?

**Review by dvoituron — CHANGES_REQUESTED** [review]
Great job !

**Review by vnbaaij — CHANGES_REQUESTED** [review]
Love it!

File: examples/Demo/Shared/Shared/DemoSearch.razor.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+using System.Diagnostics;
+using Microsoft.AspNetCore.Components;
+
+namespace FluentUI.Demo.Shared.Shared;
+
+public partial class DemoSearch
+{
+    [Inject] protected DemoNavProvider NavProvider { get; set; } = default!;
+    [Inject] protected NavigationManager NavigationManager { get; set; } = default!;
```
[the remark this answers is not in the document]
Comment (Hona) [reply examples/Demo/Shared/Shared/DemoSearch.razor.cs]: Done - used to my IDE auto formatting haha

**Review by vnbaaij — APPROVED** [review]

**Review by dvoituron — APPROVED** [review]

**Comment by dvoituron** [comment]
@Hona It's a pity that the search component doesn't work with keyboard arrows. This is a problem for our Accessibility rules. Would it be possible to make another PR and replace it with a FluentAutocomplete, for example? We'll have to see beforehand if it works in the same way :-)

image

**Comment by dvoituron** [comment]
Probably because you used a FluentSearch and a FluentMenu. And pressing Down/Up keys don't select an Items. The FluentAutocomplete do that.

**Pull request merged**